cfme/utils/log_validator.py

```diff
@@ -101,3 +101,12 @@ def validate():
             except Failed:
                 return False
         wait_for(validate, delay=delay, num_sec=num_sec, message=message, **kwargs)
+
+    def match_count(self):
```

Comment:
As it stands this function is a bit separate from the other utilities of the class. I.e. it is not making use of `_check_skip_logs`, `_check_fail_logs`, `_check_match_logs`, or `_verify_match_logs`. 

Perhaps, there can be a modification to `self.matches` to include not only the Boolean of whether or not the pattern was found, but also the count. Or there can be a separate dictionary defined in the class, `match_count` or something similar.

What do you think? My main concern is that this function is lacking some of the features that the typical `LogValidator` has already.

Reply:
Hey @john-dupuy 
- I would like to keep it as a different method in ```LogValidator``` as it returns count of matches in logs.
- If we tried to modify the self.matches dict then again we will need to add new method to get match count. Because the purpose of ```_check_match_logs``` is to check if logs are matched or not.
- Modifying existing methods will created problem while returning values and it will affect other tests too.
- Please let me know, if we need any updates in existing method - ```match_count```
- Also @digitronik is also working on it. Please have a look: https://github.com/ManageIQ/integration_tests/pull/8918

Reply:
@ganeshhubale 

Thanks for the detailed reply. Modifying `self.matches` was just a suggestion, and I agree that it will require a larger refactor that isn't necessarily warranted. 

My main concern about this function is that it doesn't support the standard features of the function, i.e. `_check_skip_logs`, `_check_fail_logs`. So, for example, if there is a line you do **not** want added to the `match_count`, you should be able to pass a skip_pattern, and those lines won't be added to the count. 

Does that make sense? I think this could be done within your existing method `match_count`, but if you'd like to do this in a different PR, that's fine with me.

Reply:
The fact is current `LogValidator` is just for matching some string pattern from a log not exactly count. `LogValidator` must be flexible about actions and cover multiple purposes. Even I am facing a problem with `tails` it's not providing me exact start with `fix_before_start` method. when I'm handling a large amount of request like `50+`. We need to overwrite it or make wrapper. My automation working properly with clearing `automation.log` but its bad practice. I'm finding a way to handle 100+ request match count. @izapolsk